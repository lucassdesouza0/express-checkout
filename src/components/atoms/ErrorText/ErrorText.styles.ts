import styled from "@emotion/styled";

export const Text = styled("h1")`
  width: 100%;
  padding: 30px 10px;
  text-align: center;
`;
