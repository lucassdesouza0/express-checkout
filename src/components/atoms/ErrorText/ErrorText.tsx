import React, { memo, ReactNode } from "react";

import { Text } from "./ErrorText.styles";

interface ErrorTextProps {
  children: ReactNode;
}

const ErrorText = ({ children }: ErrorTextProps) => {
  return <Text>{children}</Text>;
};

export default memo(ErrorText);
