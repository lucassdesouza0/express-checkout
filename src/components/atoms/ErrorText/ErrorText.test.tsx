import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRender from "../../../tests/render";

import ErrorText from "./ErrorText";

describe("Button", () => {
  const text = "404 - Page Not Found";
  test("it should mount", () => {
    testRender(<ErrorText>{text}</ErrorText>);

    const button = screen.getByRole("h1");

    expect(button).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<ErrorText>{text}</ErrorText>);
    expect(asFragment()).toMatchSnapshot();
  });
});
