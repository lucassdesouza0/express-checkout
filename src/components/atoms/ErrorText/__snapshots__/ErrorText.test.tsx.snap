// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="css-1oj4qtz-Text eyoz9500"
  >
    404 - Page Not Found
  </h1>
</DocumentFragment>
`;
